kyo-prelude/shared/src/main/scala/kyo/Abort.scala

```diff
@@ -195,6 +195,62 @@ object Abort:
       */
     inline def run[E]: RunOps[E] = RunOps(())
 
+    final class RunPartialOps[E >: Nothing](dummy: Unit) extends AnyVal:
+        /** Runs an Abort effect, converting it to a partial Result.
+          *
+          * @param v
+          *   The computation to run
+          * @tparam A
+          *   The success type of the computation
+          * @tparam S
+          *   The effect type of the computation
+          * @tparam ER
+          *   Any remaining Abort effects after running this one
+          * @return
+          *   A Result containing either the success value or the failure value, wrapped in the remaining effects
+          */
+        def apply[A: Flat, S, ER](v: => A < (Abort[E | ER] & S))(
+            using
+            frame: Frame,
+            ct: SafeClassTag[E],
+            f2: Flat[Result.Partial[E, A]],
+            reduce: Reducible[Abort[ER]]
+        ): Result.Partial[E, A] < (S & reduce.SReduced) =
+            reduce {
+                ArrowEffect.handleCatching[
+                    Const[Error[E]],
+                    Const[Unit],
+                    Abort[E],
+                    Result[E, A],
+                    Result.Partial[E, A],
+                    Abort[ER] & S,
+                    Abort[ER] & S,
+                    Any
+                ](
+                    erasedTag[E],
+                    v.map(Result.Success(_))
+                )(
+                    accept = [C] =>
+                        input =>
+                            input.isPanic ||
+                                input.asInstanceOf[Error[Any]].failure.exists(ct.accepts),
+                    handle = [C] =>
+                        (input, _) => input,
+                    done =
+                        case Panic(ct(fail))                        => Result.Failure(fail)
+                        case Panic(e)                               => throw e
+                        case error: Result.Partial[E, A] @unchecked => error,
+                    recover = e => throw e
```

Comment:
I don't love how this means the `Panic` value will be thrown, caught, and then thrown again (`done` -> `recover`), but I couldn't figure out another way to handle this. Would it be better to just `run` and then call `Abort.panic` on the `Panic` case?

Reply:
Couldn't it be `recover = Abort.panic(_)`?